haiku/src/lib.rs

```diff
@@ -27,3 +50,19 @@ pub fn get_timezone() -> Option<String> {
         s => Some(std::str::from_utf8(s).ok()?.to_owned()),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    #[cfg(not(target_os = "haiku"))]
+    fn test_fallback_on_non_haiku_platforms() {
+        assert!(super::get_timezone().is_none());
+    }
+
+    #[test]
+    #[cfg(target_os = "haiku")]
+    fn test_retrieve_time_zone_on_haiku_platforms() {
+        let timezone = super::get_timezone().unwrap();
```

Comment:
I'm not sure if we should keep this test since the function is technically (and documented to be) fallible.